### DISCUSSION TITLE
PHP 8.1 Support on latest

### DISCUSSION THREAD

**Discussion by boelter-ecomm-it** [deployphp/deployer#3931] [Bugs] [answered]
### Deployer Version

dev-master

### Target OS

Ubuntu 20.04.5 LTS

### Which PHP version are you using?

PHP 8.1

### Steps to reproduce

During our deployment process i've been pointing to dev-master to get the latest recipe changes. Today i'm getting this error when requiring the project. It seems there is an incompatibility error with 8.1. Should  php 8.1 be supported?  
 
 Running command composer global require deployer/deployer:dev-master --no-ansi --no-interaction
Changed current directory to /composer
./composer.json has been created
Running composer update deployer/deployer
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - symfony/console[v7.1.0, ..., v7.1.5] require php >=8.2 -> your php version (8.1.1) does not satisfy that requirement.
    - deployer/deployer dev-master requires symfony/console ^7.1 -> satisfiable by symfony/console[v7.1.0, ..., v7.1.5].
    - Root composer.json requires deployer/deployer dev-master -> satisfiable by deployer/deployer[dev-master].

Installation failed, deleting ./composer.json.

**Comment by boelter-ecomm-it** [comment]
Symfony/console 7.1 requires php >=8.2:
 
https://github.com/symfony/console/blob/7.1/composer.json

**Comment by boelter-ecomm-it** [comment]
Deployer composer file says php ^8.0 supported:

![image](https://github.com/user-attachments/assets/4e1db604-2eae-438f-9f19-040af28ed36e)

**Comment by boelter-ecomm-it** [comment]
I moved to use  deployer/deployer:7.5.3 instead of dev-master and that worked for me. The reason i was on dev master was because i needed some of the latest changes in the Magento recipe and was waiting for a release. Thanks!

**Comment by antonmedv** [comment, marked as answer]
dev-master version is for development. Could be broken during development.
